modules/compiler/builtins/cmake/FindBuiltinsTools.cmake

```diff
@@ -22,93 +22,109 @@ set(BUILTINS_TOOLS_CMAKE INCLUDED)
+      COMMAND ${BUILTINS_LLVM_CONFIG} --cmakedir
+      OUTPUT_VARIABLE BUILTINS_LLVM_CMAKE
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
 
-      set(BUILTINS_LLVM_CMAKE ${BUILTINS_LLVM_CMAKE}/LLVMConfig.cmake)
-      if(EXISTS ${BUILTINS_LLVM_CMAKE})
-        file(STRINGS ${BUILTINS_LLVM_CMAKE} builtins_llvm_config
-          REGEX "LLVM_ENABLE_ZLIB (1|ON|YES|TRUE|Y|0|OFF|NO|FALSE|N)")
-        foreach(line ${builtins_llvm_config})
-          string(STRIP "${line}" line)
-          string(REGEX MATCHALL
-            "^set\\(LLVM_ENABLE_ZLIB (ON|OFF|0|1)\\)$" _ "${line}")
-          set(BUILTINS_COMPILER_ZLIB_ENABLED ${CMAKE_MATCH_1})
-        endforeach()
-        if(BUILTINS_COMPILER_ZLIB_ENABLED)
-          message(FATAL_ERROR "builtins compiler will generate zlib-compressed "
-            "bitcode, but the target compiler does not have zlib support")
-            return()
-        endif()
-      else()
-        message(
-          WARNING
-          "builtins tools installation does not have necessary cmake modules; "
-          "cannot determine whether the builtins compiler generates "
-          "zlib-compressed bitcode"
-        )
+    set(BUILTINS_LLVM_CMAKE ${BUILTINS_LLVM_CMAKE}/LLVMConfig.cmake)
+    if(EXISTS ${BUILTINS_LLVM_CMAKE})
+      file(STRINGS ${BUILTINS_LLVM_CMAKE} builtins_llvm_config
+        REGEX "LLVM_ENABLE_ZLIB (1|ON|YES|TRUE|Y|0|OFF|NO|FALSE|N)")
+      foreach(line ${builtins_llvm_config})
+        string(STRIP "${line}" line)
+        string(REGEX MATCHALL
+          "^set\\(LLVM_ENABLE_ZLIB (ON|OFF|0|1)\\)$" _ "${line}")
+        set(BUILTINS_COMPILER_ZLIB_ENABLED ${CMAKE_MATCH_1})
+      endforeach()
+      if(BUILTINS_COMPILER_ZLIB_ENABLED)
+        message(FATAL_ERROR "builtins compiler will generate zlib-compressed "
+          "bitcode, but the target compiler does not have zlib support")
+          return()
       endif()
+    else()
+      message(
+        WARNING
+        "builtins tools installation does not have necessary cmake modules; "
+        "cannot determine whether the builtins compiler generates "
+        "zlib-compressed bitcode"
+      )
     endif()
+  endif()
 
-    string(REGEX MATCH ".*clang${CA_HOST_EXECUTABLE_SUFFIX}$"
-      COMPILER_RESULT ${BUILTINS_COMPILER})
-    if(NOT "${BUILTINS_COMPILER}" STREQUAL "${COMPILER_RESULT}")
-      message(FATAL_ERROR
-        "Clang is the only supported compiler '${BUILTINS_COMPILER}' is not "
-        "supported. Clean the build directory and set BUILTINS_COMPILER to "
-        "'<llvm-build>/bin/clang${CA_HOST_EXECUTABLE_SUFFIX}' to fix this "
-        "error.")
-    endif()
-    if(NOT EXISTS ${COMPILER_RESULT})
-      message(FATAL_ERROR
-        "${COMPILER_RESULT} does not exist. Clean the build directory and set "
-        "CA_BUILTINS_TOOLS_DIR to '<llvm-build>' or BUILTINS_COMPILER to "
-        "'<llvm-build>/bin/clang${EXTENSION}' to fix this error.")
-    endif()
+  string(REGEX MATCH ".*clang${CA_HOST_EXECUTABLE_SUFFIX}$"
```

Comment:
I realize you did not change this beyond the indentation, but this seems wrong. `CA_HOST_EXECUTABLE_SUFFIX` would be expected to be something like `.exe`, and we don't want to interpret the `.` in there as a wildcard character. Could we go over the touched CMake bits to see if there's more that doesn't make sense and could be fixed while we're here anyway?

Reply:
Yeah, good point. But I don't really understand the point of his regex matching at all. I can't see a way that `BUILTINS_COMPILER` is anything but `${tools_dir}/clang${CA_HOST_EXECUTABLE_SUFFIX}` so isn't this regex trivially always matching? Or am I misunderstanding? Can the user set `BUILTINS_COMIPLER` themselves or something?

Reply:
Hmm, I think you're right. I wonder if at one point this was planned to be a cache variable (which the user can adjust) but I cannot find it ever having been one. That allows for a nice cleanup.

Reply:
Yeah, that's a pleasant surprise. Thanks!